### PR TITLE
Ticket2994 keithley drift value c script

### PR DESCRIPTION
**Pull request by esouthren**

### CODE REVIEW

File: Keithley_2700Sup/calculate_drift.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	// If we have a new timestamped value, calculate the drift
+	if (time_delta > 0) {
+		double change_over_time = 0;
+		previous_drift = previous_drift * 0.98;
+		
+		// If there is previous temperatura data
+		if (previous_temp > 0.0000001) {
+			change_over_time = ((temp_delta/time_delta)*60);
+		}
+		change_over_time = change_over_time/50;
```
Comment (AdrianPotter) [inline Keithley_2700Sup/calculate_drift.c]: This alorithm is full of magic numbers. I assume `0.98` comes from `1-1/50`? Not clear how this is working. Please address as in the OPI PR

File: Keithley_2700Sup/calculate_drift.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+		// If change_over_time and previous_drift are valid values (not infinity or NaN)
+		if ((change_over_time == change_over_time) && (previous_drift != previous_drift)) { // checking for NaN
```
Comment (AdrianPotter) [inline Keithley_2700Sup/calculate_drift.c]: Can you use `isNan` here? http://www.cplusplus.com/reference/cmath/isnan/. I'm a Python programmer by heart so this kind of magic logic statement feels very unatural and error prone.

**Review by AdrianPotter — CHANGES_REQUESTED** [review]

File: Keithley_2700Sup/calculate_drift.c (hunk 2 above)
Comment (esouthren) [reply Keithley_2700Sup/calculate_drift.c]: Added NaN check

File: Keithley_2700Sup/drift_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    previous_drift = previous_drift * previous_proportion;
+
+    // If there is previous temperature data
+    if (prev_temp > 0) {
+        temp_change_over_time = ((temp_delta/time_delta)*60); // seconds in minute
+    }
+    temp_change_over_time = temp_change_over_time/SMOOTHING_FACTOR;
+
+    // If temp_change_over_time and previous_drift are valid values (not infinity or NaN)
+    if ((!isnan(temp_change_over_time)) && (isnan(previous_drift))) {
```
Comment (ghost) [inline Keithley_2700Sup/drift_utils.cpp:25]: Be explicit about why this is being done and add it to the wiki for future prosperity

Comment (LiamPanchaud) [reply Keithley_2700Sup/drift_utils.cpp:25]: Added comment to explain, and included in wiki page

**Pull request merged**